src/main/EncryptorDecryptor.java

```diff
@@ -0,0 +1,96 @@
+package main;
+
+import main.algorithms.Algorithm;
+import main.algorithms.ShiftAlgorithmImpl;
+import main.algorithms.UnicodeAlgorithmImpl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class EncryptorDecryptor {
+    private String mode;
+    private String data;
+    private int key;
+    private String in;
+    private String out;
+    private Algorithm algorithm;
+
+    public EncryptorDecryptor() {
+        mode = "enc";
+        data = "";
+        key = 0;
+        out = "";
+    }
+
+
+    public void invoke() {
+        readFromFile();
+        String result = switch (this.mode) {
+            case "enc" -> algorithm.encrypt(data, key);
+            case "dec" -> algorithm.decrypt(data, key);
+            default -> "";
+        };
+
+        writeToFile(result);
+    }
+
+
+    private void readFromFile() {
+        if (data.equals("") && in != null) {
+            try {
+                data = new String(Files.readAllBytes(Paths.get(in)));
+            } catch (IOException e) {
+                System.err.println("Error reading from file");
+                System.exit(1);
+            }
+        }
+    }
+
+    private void writeToFile(String result) {
+        if (out.equals("")) System.out.println(result);
```

Comment:
Please use **out.isEmpty()**

Reply:
The readability is better when you use curly braces. Can you please change it to this:
```
if (out.equals("")) {
    System.out.println(result);
}  else {
    ...
}
```

Reply:
Fixed.